src/main/scala/inox/solvers/theories/MapMergeEncoder.scala

```diff
@@ -0,0 +1,66 @@
+/* Copyright 2009-2020 EPFL, Lausanne */
+
+package inox
+package solvers
+package theories
+
+trait MapMergeEncoder extends SimpleEncoder {
+  import trees._
+
+  object MapMergeEncoded {
```

Comment:
One small improvement in the encoder would be to hide the choose below a `FunDef`. This would help the solvers prove programs involving `MapMerge` and also make the decoding step easier.